tests/unit/test_provider_manifest.py

```diff
@@ -55,3 +56,12 @@ def test_provider_manifest_invalid_certificate(provider_manifest):
         provider_manifest.evaluate()
         == "Certificate 'keystone-ssl-ca' is not valid PEM certificate."
     )
+
+
+def test_apply_resources(provider_manifest):
+    provider_manifest.apply_manifests()
+    provider_manifest.client.patch.assert_called_once()
+    args, kwargs = provider_manifest.client.patch.call_args
+    assert args[:2] == (Deployment, "k8s-keystone-auth")
+    assert args[2].spec.template.metadata.annotations["juju.io/restartedAt"]
+    assert kwargs == {"namespace": "kube-system"}
```

Comment:
Can we check for ready/running here?

Reply:
in the unit tests?  not really

Reply:
Just keeping you on your toes 😄 But my mistake... carry on!